src/pages/RepoPage/RepoPageTabs.spec.tsx

```diff
@@ -1009,4 +1009,79 @@ describe('useRepoTabs', () => {
       })
     })
   })
+
+  describe('tests tab', () => {
+    describe('when test analytics is enabled', () => {
+      it('adds the tests link to the array', async () => {
+        setup({ testAnalyticsEnabled: true })
+        const { result } = renderHook(
+          () =>
+            useRepoTabs({
+              refetchEnabled: false,
+            }),
+          { wrapper: wrapper('/gh/codecov/test-repo') }
+        )
+
+        const expectedTab = [
+          {
+            pageName: 'failedTests',
+            children: expect.anything(),
+          },
+        ]
+        await waitFor(() =>
+          expect(result.current).toEqual(expect.arrayContaining(expectedTab))
+        )
+      })
+    })
+
+    describe('when test analytics is disabled', () => {
+      it('does not add the tests link to the array', async () => {
+        setup({ testAnalyticsEnabled: false })
+        const { result } = renderHook(
+          () =>
+            useRepoTabs({
+              refetchEnabled: false,
+            }),
+          { wrapper: wrapper('/gh/codecov/test-repo') }
+        )
+
+        const expectedTab = [
+          {
+            pageName: 'failedTestsOnboarding',
+            children: expect.anything(),
+          },
+        ]
+
+        await waitFor(() =>
+          expect(result.current).toEqual(expect.arrayContaining(expectedTab))
+        )
+      })
+    })
+
+    describe('when test analytics is disabled and user is not part of the org', () => {
+      it('does not add the tests link to the array', async () => {
+        setup({ testAnalyticsEnabled: false, isCurrentUserPartOfOrg: false })
+        const { result } = renderHook(
+          () =>
+            useRepoTabs({
+              refetchEnabled: false,
+            }),
+          { wrapper: wrapper('/gh/codecov/test-repo') }
+        )
+
+        const expectedTab = [
+          {
+            pageName: 'failedTestsOnboarding',
+            children: expect.anything(),
+          },
+        ]
+
+        await waitFor(() =>
+          expect(result.current).not.toEqual(
```

Comment:
Should we put what is expected here rather than .not() and have the variable name "expectedTab?" Would reduce cognitive burden here imo

Reply:
Sure i will add that as part of the test.

I'm specifically checking for the tab's existence because if analytics isn't enabled, it shouldn't appear—unless it's the specific case being tested. It's a bit more unique to this scenario.

Reply:
Also nice catch for the variable name!